File: wrappers/java/src/main/java/org/hyperledger/indy/sdk/ParamGuard.java
package org.hyperledger.indy.sdk;

public class ParamGuard {

	public static void notNull(Object param, String paramName) {
		if(param == null)
			throw new IllegalArgumentException("A value must be provided for the '" + paramName + "' parameter.");
	}
	
	public static void notNullOrWhiteSpace(String param, String paramName) {
		if(StringUtils.isNullOrWhiteSpace(param))
			throw new IllegalArgumentException("A non-empty string must be provided for the '" + paramName + "' parameter.");
	}
}

File: wrappers/dotnet/indy-sdk-dotnet/Utils/ParamGuard.cs
using System;

namespace Hyperledger.Indy.Utils
{
    /// <summary>
    /// Common guards.
    /// </summary>
    internal static class ParamGuard
    {
        /// <summary>
        /// Guards against null values.
        /// </summary>
        /// <param name="param">The parameter to check.</param>
        /// <param name="paramName">The name of the parameter.</param>
        /// <exception cref="ArgumentNullException">Thrown if param was null.</exception>
        public static void NotNull(object param, string paramName)
        {
            if (param == null)
                throw new ArgumentNullException(paramName);
        }

        /// <summary>
        /// Guards against null and strings containing nothing but whitespace.
        /// </summary>
        /// <param name="param">The parameter to check.</param>
        /// <param name="paramName">The name of the parameter.</param>
        /// <exception cref="ArgumentException">Thrown if param was null or contained whitespace.</exception>
        public static void NotNullOrWhiteSpace(string param, string paramName)
        {
            if (string.IsNullOrWhiteSpace(param))
                throw new ArgumentException("A value must be provided.", paramName);
        }
    }
}
